File: src/main/java/com/vjiki/music/dto/PlaylistResponse.java
package com.vjiki.music.dto;

import java.time.OffsetDateTime;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PlaylistResponse {
    private UUID id;
    private UUID userId;
    private String userName;
    private String userNickname;
    private String name;
    private String description;
    private String coverUrl;
    private String type;

    @JsonProperty("isPublic")
    private Boolean isPublic;

    private OffsetDateTime createdAt;
    private OffsetDateTime modifiedAt;
}

File: bin/main/com/vjiki/music/dto/PlaylistResponse.kt
package com.vjiki.music.dto

import java.time.OffsetDateTime
import java.util.UUID

data class PlaylistResponse(
    val id: UUID,
    val userId: UUID,
    val userName: String?,
    val userNickname: String?,
    val name: String,
    val description: String?,
    val coverUrl: String?,
    val type: String,
    val isPublic: Boolean,
    val createdAt: OffsetDateTime?,
    val modifiedAt: OffsetDateTime?
)

